Doc 1776/open banking partner documentation on docsswanio (#641)
swan-io/docs: docs/topics/open-banking/index.mdx

:::note TPPs don't connect to Swan directly
All requests go through Salt Edge, which acts as the compliance layer between TPPs and Swan.
:::

## Architecture {#architecture}

```mermaid
%%{ init: { 'flowchart': { 'curve': 'linear' } } }%%
flowchart TB
    U([User]) --> SA([Software application])
    SA --> TPP([TPP aggregator])
    TPP --> SE([Salt Edge<br/>PSD2 compliance platform])
    SE --> PC([PSD2 Connector<br/>Swan])
    PC --> API([Swan GraphQL API<br/>and core systems])
```

| Component | Role |
|---|---|
| Software application | The application the end user interacts with, such as accounting, finance, or ERP tools. |
| TPP | Regulated aggregator the software works with to access bank data or initiate payments. |
| Salt Edge | PSD2 compliance platform that manages TPP registration, authentication flows, and data formatting to the Berlin Group standard. |
| PSD2 Connector | Integration layer between Salt Edge and Swan's core systems. |
| Swan GraphQL API | Source of account data and payment execution. |

## Ecosystem {#ecosystem}

Open Banking relies on a network of regulated TPPs that connect banking data to software applications.

### Connected TPPs {#ecosystem-tpps}

> :::note

Reviewer: This needs a custom title - all call-outs need one.

Author: Snapshot date mismatch with the software table below ("as of May 2026" vs "as of July 2026") is intentional for now, the May date reflects when the TPP list was captured, and confirming it's still current needs Martin. Parked on [DOC-1776](https://linear.app/swan/issue/DOC-1776/open-banking-partner-documentation-on-docsswanio#comment-3e512476) for his return; not blocking.

|---|---|---|---|
| User consent token | 180 days | TPP and Salt Edge | Grants the TPP access to account data. Requires user SCA to renew. |
| Technical refresh token | 24 hours | Swan and Salt Edge | Maintains the data refresh connection. Renewed automatically. |

:::note User token renewal
Every 180 days, the user must re-authenticate with SCA to renew the consent token.
PSD2 requires this.
Renewal is initiated by the TPP through Salt Edge.
Swan cannot trigger this renewal directly.
:::

:::note Technical token refresh
Most TPPs can perform up to 4 refreshes per day, see the [PSD2 EBA Q&A on refresh frequency](https://www.eba.europa.eu/single-rule-book-qa/qna/view/publicId/2018_4210).
:::

### Consent validity {#consent-validity}

- **AIS**: one consent grants data access for up to 180 days, then requires re-authentication.
- **PIS**: each payment requires its own consent.

:::note Transactions since account creation
The 180-day limit applies to how long the consent grants data access, not to the time range of transactions you can view.
By default, Swan returns all transactions since the account was created.
:::

### Consent revocation {#consent-revocation}

> The user consent token can be revoked by the TPP, following a request from the end user.

Reviewer: ```suggestion
Users can ask their TPP to revoke the consent token. Swan can't revoke it directly.
```
Active

:::tip Under PSD3
With PSD3 (the third Payment Services Directive), allowing end users to revoke their consent directly from their online banking interface will become mandatory.
:::

## Registration and testing {#testing}

To get access, TPPs first apply to Salt Edge at [compliance-hub@saltedge.com](mailto:compliance-hub@saltedge.com).
Salt Edge reviews the application and, if approved, grants access to their platform and Swan's PSD2 API.
Testing happens in Salt Edge's own sandbox, not in Swan's Sandbox environment.

> ## Key concepts {#key-concepts}

Reviewer: These are already at the top of the page - is it necessary to repeat?
